sugondat/shim/src/cli.rs

```diff
@@ -87,6 +87,14 @@ pub struct SugondatRpcParams {
     /// The address of the sugondat-node to connect to.
     #[clap(long, default_value = "ws://localhost:9988", env = ENV_SUGONDAT_NODE_URL)]
     pub node_url: String,
+
+    /// By default, the connection to the sugondat-node is persistent.
+    /// If, for any reason, it's impossible to connect to the sugondato-node
+    /// or if it fails while connected, it retries to reestablish the connection.
+    ///
+    /// This flag avoids this behavior by attempting only a single connection
```

Comment:
Several notes: 

1. I am not sure if it's fair to mention the connection persistence, since that only matters after the first connection was established.
2. the last line would read better as "by attempting to connect only once".

Reply:
> I am not sure if it's fair to mention the connection persistence, since that only matters after the first connection was established.

I named it persistent because I was convinced that it remains persistent, not just during the first connection. In the `struct client`, there are loops that continue until a proper response is received from the node. 

Let's take for example the code of `block_hash`

```rust
    pub async fn block_hash(&self, height: u64) -> anyhow::Result<Option<H256>> {
        loop {
            let conn = self.connector.ensure_connected().await;
            let block_hash: Option<H256> = match conn
                .raw
                .request("chain_getBlockHash", rpc_params![height])
                .await // <--
            {
                Ok(it) => it,
                Err(err) => {
                    tracing::error!(?err, "failed to query block hash");
                    self.connector.reset().await; // <--
                    continue; 
                }
            };

            break match block_hash {
                None => Ok(None),
                Some(h) if h == H256::zero() => {
                    // Little known fact: the sugondat node returns a zero block hash if there is no block
                    // at the given height.
                    Ok(None)
                }
                Some(block_hash) => Ok(Some(block_hash)),
            };
        }
    }
```

In the code above if the rpc request fails, and one reason could be the connection dropped (*), the connection is reset, spawning a new connection task, without going back to the coller of the function. That's why I think it is persistant

\* probably here I'm wrong and the rpc request can't fails do tue a connection problem?

> the last line would read better as "by attempting to connect only once".

Ok!

Reply:
Persistent, to me, implies that the connection is always actively maintained.

However, the retries are only attempted in response to a request. I believe it would be possible for the connection to drop for extended periods of time and then be reconnected. It would not be correct to call that a persistent connection.